spec/models/allocation_spec.rb

```diff
@@ -32,16 +32,59 @@
     end
   end
 
-  describe '#refresh_moves_count!' do
-    let(:cancelled_move) { create :move, :cancelled }
-    let(:proposed_move) { create :move, :proposed }
-    let(:requested_move) { create :move, :requested }
-    let(:completed_move) { create :move, :completed }
-    let(:moves) { [cancelled_move, proposed_move, requested_move, completed_move] }
-    let!(:allocation) { create :allocation, moves: moves, moves_count: 1 }
+  describe '#refresh_status_and_moves_count!' do
+    context 'when updating moves count' do
+      let(:cancelled_move) { create :move, :cancelled }
+      let(:proposed_move) { create :move, :proposed }
+      let(:requested_move) { create :move, :requested }
+      let(:completed_move) { create :move, :completed }
+      let(:moves) { [cancelled_move, proposed_move, requested_move, completed_move] }
+      let!(:allocation) { create :allocation, moves: moves, moves_count: 1 }
+
+      it 'updates the number of non cancelled moves' do
+        expect { allocation.refresh_status_and_moves_count! }.to change { allocation.reload.moves_count }.from(1).to(3)
```

Comment:
I don't think(??) there is any need to the `reload` in this statement

Reply:
I suspect it is - I've used a similar pattern in recent specs and without this rspec doesn't pick up the change to the model. Good to test this though :)

Reply:
I think we should be using it here (and elsewhere in this test), as we are persisting the value, not just amending the attributes on the fly, and those aren't picked up sometimes

Reply:
how about instead, one extra test to check for persistence (using reload), and then remove reload from these tests? by keeping it here there is an implication that the record must always be reloaded from the database for it to be correct, which doesn't seem like a good idea.

Reply:
So I've revisited this, as initially when I wrote tests there were random failures but that might have been related to the way it was implemented previously. I've removed them all and kept one reload now where the test fails otherwise.